frontend/app/controllers/Healthcheck.scala

```diff
@@ -1,34 +1,56 @@
 package controllers
 
-import com.github.nscala_time.time.Imports._
+import com.gu.membership.model.{FriendTierPlan, PaidTierPlan, StaffPlan}
+import com.gu.membership.salesforce.Tier.{Partner, Patron, Supporter}
 import com.gu.monitoring.CloudWatchHealth
 import play.api.Logger
+import play.api.libs.concurrent.Execution.Implicits._
 import play.api.mvc.{Action, Controller}
 import services.{GuardianLiveEventService, TouchpointBackend}
 
-case class Test(name: String, ok: () => Boolean)
+import scala.concurrent.Await
+import scala.util.{Try, Failure, Success}
+
+case class Test(name: String, exec: () => Boolean) {
+  def ok() = {
+    val passed = exec()
+    if (!passed) Logger.warn(s"Test $name failed, health check will fail")
+    passed
+  }
+}
+
+object ZuoraPing {
+  import com.github.nscala_time.time.Imports._
+  val zuoraSoapService = TouchpointBackend.Normal.zuoraSoapService
+  def ping = zuoraSoapService.lastPingTimeWithin(2.minutes)
+}
 
 object Healthcheck extends Controller {
-  val zuoraService = TouchpointBackend.Normal.zuoraSoapService
+  import scala.concurrent.duration._
+
+  val zuoraSoapService = TouchpointBackend.Normal.zuoraSoapService
+  val subscriptionService = TouchpointBackend.Normal.subscriptionService
 
-  val tests = Seq(
+  val productRatePlanTiers = List(FriendTierPlan, StaffPlan,
+    PaidTierPlan(Supporter, true), PaidTierPlan(Supporter, false),
+    PaidTierPlan(Partner, true), PaidTierPlan(Partner, false),
+    PaidTierPlan(Patron, true), PaidTierPlan(Patron, false))
+
+  def tests = Seq(
     Test("Events", () => GuardianLiveEventService.events.nonEmpty),
     Test("CloudWatch", () => CloudWatchHealth.hasPushedMetricSuccessfully),
-    Test("Zuora", () => zuoraService.lastPingTimeWithin(2.minutes))
-  )
-
-  def healthcheck() = Action { req =>
-    val failedTests = tests.filterNot(_.ok())
+    Test("ZuoraPing", () => ZuoraPing.ping)) ++
+    productRatePlanTiers.map(tier => Test(s"ZuoraCatalog: $tier", () =>
+      Try(Await.result(subscriptionService.tierRatePlanId(tier).map(_ => true), 10 millis)) match {
```

Comment:
:-1: to calling Await in the healthcheck :-(  I prefer that the healthcheck calls `.value` on Futures we need to check (which is non-blocking, just getting the current value out of the future), tho' it looks like we might need a bit of a refactor to do that here.